packages/typespec-python/package.json

```diff
@@ -46,6 +46,7 @@
     "@azure-tools/typespec-azure-resource-manager": ">=0.42.0 <1.0.0",
     "@azure-tools/typespec-autorest": ">=0.42.0 <1.0.0",
     "@azure-tools/typespec-client-generator-core": ">=0.42.3 <1.0.0",
+    "@azure-tools/codegen": ">=2.10.0 <3.0.0",
```

Comment:
this lib i believe is from autorest toolchain. i doubt if we could continue use it?

Reply:
Is there any concern about the package? @iscai-msft

Reply:
Yes, I agree we can't use it.

Reply:
OK. Let me copy the code.

Reply:
Update: I find it is too hard to copy the code since the logic is more complicated than I estimate. 
Another solution is that we find a 3rd party package while I am not sure whether it meets our requirements and I think it may bring more risks compared with this package.
I choose the package because it is also used in m4 which can reduce the naming breaking if we migrate swagger to Typespec in the future. So I still hope we could use the package to fix the naming issue.

FYI: java emitter is also using the package now.

Reply:
we will have to copy it at some point. What makes copying the code complicated?

Reply:
@iscai-msft At first I think only need to copy function A, while then A depends on B of B.js, then B depends on C of C.js then C depends on D of D.js, then... Before I finish all, it already totally costs over hundreds of lines and has some lint warning. 
If it is strictly forbidden to use the package, I will have to take more time to make some code change instead of easy copy-paste.

Reply:
Let's hold the PR until timotheeguerin gets back.

Reply:
Hey sorry just realized we are already using `change-case` all over including in tcgc already, use that

Reply:
I see and will make another PR to try `change-case`